src/commonMain/kotlin/exchange.dydx.abacus/processor/wallet/account/AccountProcessor.kt

```diff
@@ -199,15 +171,14 @@ internal class V4AccountProcessor(
     private val delegationsProcessor = AccountDelegationsProcessor(parser)
     private val tradingRewardsProcessor = AccountTradingRewardsProcessor(parser)
     private val launchIncentivePointsProcessor = LaunchIncentivePointsProcessor(parser)
+    private val stakingRewardsProcessor = StakingRewardsProcessor(parser)
+    private val unbondingProcessor = DelegationUnbondingProcessor(parser)
 
     internal fun processAccountBalances(
         existing: InternalAccountState,
         payload: List<OnChainAccountBalanceObject>?,
     ): InternalAccountState {
-        val balances = balancesProcessor.process(existing.balances, payload)
-        if (balances != existing.balances) {
-            existing.balances = balances
-        }
+        existing.balances = balancesProcessor.process(existing.balances, payload)
```

Comment:
👍

Reply:
[suggestion] honestly this is so clean now i wonder if we even need a function for this? could just be an inline call.

if we do decide to do it, am in favor of doing it in a followup PR

Reply:
We can potentially reduce the hierarchy of the processors in the future, but for now I think it's better to keep the existing hierarchy so that we have a one-to-one mapping between the old parsers and new ones.

Also, it can be argued that although the hierarchy of processors creates lots of boilerplate code, it makes the code better structured (and hopefully easier to maintain once you understand the structure)

Reply:
yeah 100% agree on not doing this now, if we even decide to do it.

fair, i think that's probably a bigger discussion to have! thanks rui